### PR TITLE
Dialog/Popup: Add aria-labelledby (T1182767)

### PR DESCRIPTION
**Pull request by marker-dao**

### CODE REVIEW

File: packages/devextreme/testing/tests/DevExpress.ui/dialog.tests.js [hunk 1]
```diff
@@ -333,6 +336,24 @@ module('dialog', {
         config({ rtlEnabled: false });
     });
 
+    test('dialog aria-labelledby is equal message id if title is null', function(assert) {
```
Comment (ksercs) [inline packages/devextreme/testing/tests/DevExpress.ui/dialog.tests.js]: [minor]
```suggestion
    test('dialog aria-labelledby is equal to message id if title is null', function(assert) {
```

File: packages/devextreme/js/ui/popup/ui.popup.js [hunk 2]
```diff
@@ -388,6 +390,17 @@ const Popup = Overlay.inherit({
         } else if(this._$title) {
             this._$title.detach();
         }
+
+        this._toggleAriaLabel();
+    },
+
+    _toggleAriaLabel() {
+        const { title, showTitle } = this.option();
+        const shouldSetAria = showTitle && !!title;
```
Comment (ksercs) [inline packages/devextreme/js/ui/popup/ui.popup.js]: [minor]
Aria is too common to me
```suggestion
        const shouldLabelTitle = showTitle && !!title;
```

File: packages/devextreme/js/ui/popup/ui.popup.js [hunk 3]
```diff
@@ -388,6 +390,17 @@ const Popup = Overlay.inherit({
...
+        const titleId = new Guid();
```
Comment (ksercs) [inline packages/devextreme/js/ui/popup/ui.popup.js]: Let's reduce copypaste below
```suggestion
        const titleId = shouldLabelTitle ? new Guid() : null;
```

File: packages/devextreme/testing/tests/DevExpress.ui.widgets/popup.tests.js [hunk 4]
```diff
@@ -169,6 +170,50 @@ QUnit.module('basic', () => {
         assert.strictEqual($overlayContent.attr('role'), 'dialog');
     });
 
+    QUnit.test('aria-labelledby is equal title id', function(assert) {
```
Comment (ksercs) [inline packages/devextreme/testing/tests/DevExpress.ui.widgets/popup.tests.js]:
```suggestion
    QUnit.test('aria-labelledby is equal to title id', function(assert) {
```
P.S. repeats below

File: packages/devextreme/js/ui/popup/ui.popup.js [hunk 5]
```diff
@@ -368,13 +371,12 @@ const Popup = Overlay.inherit({
 
     _renderTitle: function() {
         const items = this._getToolbarItems('top');
-        const titleText = this.option('title');
-        const showTitle = this.option('showTitle');
+        const { title, showTitle } = this.option();
 
-        if(showTitle && !!titleText) {
+        if(showTitle && !!title) {
```
Comment (alexanderPolosatov) [inline packages/devextreme/js/ui/popup/ui.popup.js:376]: may be replace _!!title_ to _Boolean(title)_?

File: packages/devextreme/testing/tests/DevExpress.ui/dialog.tests.js (hunk 1 above)
Comment (marker-dao) [reply packages/devextreme/testing/tests/DevExpress.ui/dialog.tests.js]: [2ea37b479fd6e58d4d2122c7b40e37e69a5268b9]

File: packages/devextreme/js/ui/popup/ui.popup.js (hunk 2 above)
Comment (marker-dao) [reply packages/devextreme/js/ui/popup/ui.popup.js]: [2ea37b479fd6e58d4d2122c7b40e37e69a5268b9]

File: packages/devextreme/js/ui/popup/ui.popup.js (hunk 3 above)
Comment (marker-dao) [reply packages/devextreme/js/ui/popup/ui.popup.js]: [2ea37b479fd6e58d4d2122c7b40e37e69a5268b9]

File: packages/devextreme/testing/tests/DevExpress.ui.widgets/popup.tests.js (hunk 4 above)
Comment (marker-dao) [reply packages/devextreme/testing/tests/DevExpress.ui.widgets/popup.tests.js]: [2ea37b479fd6e58d4d2122c7b40e37e69a5268b9]

File: packages/devextreme/testing/tests/DevExpress.ui.widgets/popup.tests.js [hunk 6]
```diff
@@ -169,6 +170,50 @@ QUnit.module('basic', () => {
...
+            visible: true,
+        }).dxPopup('instance');
+
+        const $overlayContent = instance.$content().parent();
+        const titleId = $overlayContent.find(`.${TOOLBAR_LABEL_CLASS}`).attr('id');
+
+        assert.strictEqual($overlayContent.attr('aria-labelledby'), titleId);
+    });
+
+    QUnit.test('aria-labelledby is not setted when title is null', function(assert) {
```
Comment (AlexanderMoiseev) [inline packages/devextreme/testing/tests/DevExpress.ui.widgets/popup.tests.js]:
```suggestion
    QUnit.test('aria-labelledby is not set when title is null', function(assert) {
```

Comment (marker-dao) [reply packages/devextreme/testing/tests/DevExpress.ui.widgets/popup.tests.js]: [291381b25db049bb2e62c8684e0f9f62f3445a7e]

File: packages/devextreme/testing/tests/DevExpress.ui.widgets/popup.tests.js [hunk 7]
```diff
@@ -169,6 +170,50 @@ QUnit.module('basic', () => {
...
+        const titleIdFirst = $overlayContent.find(`.${TOOLBAR_LABEL_CLASS}`).attr('id');
+
+        assert.strictEqual($overlayContent.attr('aria-labelledby'), titleIdFirst);
+
+        instance.option('title', 'title 2');
+        const titleIdSecond = $overlayContent.find(`.${TOOLBAR_LABEL_CLASS}`).attr('id');
+
+        assert.strictEqual($overlayContent.attr('aria-labelledby'), titleIdSecond);
+        assert.notStrictEqual($overlayContent.attr('aria-labelledby'), titleIdFirst);
+
```
Comment (AlexanderMoiseev) [inline packages/devextreme/testing/tests/DevExpress.ui.widgets/popup.tests.js:209]: [minor] just a sug
```suggestion
        const secondTitle = 'title2';
        instance.option('title', secondTitle);
        assert.strictEqual($overlayContent.attr('aria-labelledby'), secondTitle);

```
i think its more clear and and we don't need notStrictEqual check

File: packages/devextreme/testing/tests/DevExpress.ui.widgets/popup.tests.js [hunk 8]
```diff
@@ -169,6 +170,50 @@ QUnit.module('basic', () => {
...
+    QUnit.test('aria-labelledby is not setted when title is null', function(assert) {
+        const instance = $('#popup').dxPopup({ visible: true }).dxPopup('instance');
+
+        const $overlayContent = instance.$content().parent();
+
+        assert.strictEqual($overlayContent.attr('aria-labelledby'), undefined);
+    });
+
+    QUnit.test('aria-labelledby is correct in runtime change', function(assert) {
```
Comment (AlexanderMoiseev) [inline packages/devextreme/testing/tests/DevExpress.ui.widgets/popup.tests.js]:
```suggestion
    QUnit.test('aria-labelledby should be correct after runtime change', function(assert) {
```

File: packages/devextreme/testing/tests/DevExpress.ui.widgets/popup.tests.js (hunk 7 above)
Comment (marker-dao) [reply packages/devextreme/testing/tests/DevExpress.ui.widgets/popup.tests.js:209]: aria-labelledby should be ID of the label element. Not a simple string.

**Review by AlexanderMoiseev — APPROVED** [review]

File: packages/devextreme/testing/tests/DevExpress.ui.widgets/popup.tests.js (hunk 8 above)
Comment (marker-dao) [reply packages/devextreme/testing/tests/DevExpress.ui.widgets/popup.tests.js]: [56f27af78057ccc8f728d48b287e8000ff534a2f]

**Review by Zedwag — APPROVED** [review]
